SetUpEnvironmentDriver.cs

```diff
@@ -0,0 +1,161 @@
+                //    sauceUrl = new Uri("http://localhost:4444/wd/hub");
+                //    SetDriver(new RemoteWebDriver(sauceUrl, options));
+                //    break;
+                case ("saucelabs"):
+                    String currentDate = DateTime.Now.ToString("yyyy-MM-dd");
+                    String currentTime = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss");
+
+                    var browserOptions = options;
+                    browserOptions.PlatformName = "Windows 10";
+                    browserOptions.BrowserVersion = null;
+
+                    var sauceOptions = new Dictionary<string, object>();
+                    sauceOptions.Add("build", "Platform Configurator Build " + currentDate);
+                    sauceOptions.Add("job", "Platform Configurator Job " + currentTime);
+                    sauceOptions.Add("username", Environment.GetEnvironmentVariable("Ivo_Igov"));
+                    sauceOptions.Add("accessKey", Environment.GetEnvironmentVariable("930c032e-9c3e-4ea4-841e-eab1aaba457e"));
+                    browserOptions.AddAdditionalOption("sauce:options", sauceOptions);
+
+                    sauceUrl = new Uri("https://Ivo_Igov:930c032e-9c3e-4ea4-841e-eab1aaba457e@ondemand.eu-central-1.saucelabs.com:443/wd/hub");
+                    SetDriver(new RemoteWebDriver(sauceUrl, browserOptions));
+
+                    ((IJavaScriptExecutor)GetDriver()).ExecuteScript($"sauce:job-name= {TestContext.CurrentContext.Test.MethodName}");
+
+                    break;
+                default:
+                    throw new ArgumentException("You need to select a valid strategy.");
+            }
+        }
+
+        public static void StopDriver()
+        {
+            var reader = new GetSettings();
+            var settings = reader.GetConfigFileData();
+            if (!settings.Environment.Equals("saucelabs"))
+            {
+                var status = TestContext.CurrentContext.Result.Outcome.Status;
+                if (status == TestStatus.Failed)
+                {
+                    string outputDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+                    Screenshot ss = ((ITakesScreenshot)GetDriver()).GetScreenshot();
+                    string currentDate = DateTime.Now.ToString().Replace("/", "_").Replace(" ", "_").Replace(":", "_");
+                    var relativePathToScreenshots = @"..\..\..\..\Screenshots";
+                    var fullPathToScreenshots = Path.GetFullPath(Path.Combine(outputDirectory, relativePathToScreenshots));
+                    string screenshotFileName = "//Screenshot{0}.png";
+
+                    ss.SaveAsFile(fullPathToScreenshots + String.Format(screenshotFileName, currentDate), ScreenshotImageFormat.Png);
+                }
+            }
+            else
+            {
+                SauceLabsClean();
+            }
+
+            Console.WriteLine($"Date and Time: {DateTime.Now}");
```

Comment:
better to use logger if you need such information. (Not in scope of this task)

Reply:
Will research logging, since I have not written loggers so far.